homeassistant/components/advantage_air/binary_sensor.py

```diff
@@ -35,15 +35,13 @@ class AdvantageAirZoneFilter(AdvantageAirEntity, BinarySensorEntity):
 
     _attr_device_class = DEVICE_CLASS_PROBLEM
 
-    @property
-    def name(self):
-        """Return the name."""
-        return f'{self._ac["name"]} Filter'
-
-    @property
-    def unique_id(self):
-        """Return a unique id."""
-        return f'{self.coordinator.data["system"]["rid"]}-{self.ac_key}-filter'
+    def __init__(self, instance, ac_key):
+        """Initialize an Advantage Air Filter."""
+        super().__init__(instance, ac_key)
+        self._attr_name = f'{self._ac["name"]} Filter'
```

Comment:
This appears to prevent the name from being updated by the source without a restart.

Reply:
Is it reasonable to expect the user to change the name from the source or should they do it in HA?

Reply:
Can be both. If overridden in Home Assistant, that will win; otherwise it will take over whatever was served from the integration.

Reply:
![Screenshot_20210714-060447.png](https://user-images.githubusercontent.com/2647914/125517672-5aff42cc-edb4-4f0d-91a7-6a5f35ec1e57.png)

While I admit this probably doesn't happen often, it is easy to rename from the source and currently it is dynamically updating in HA.

Reply:
```
        self._attr_name = f'{self.coordinator.data["aircons"][self.ac_key]["info"]["name"]} Filter'
```
This thought this might work instead, but it also doesnt update, so it seems the _attr_name is getting set by value.

Reply:
I think it is fine to keep it as placed right now. It stops re-calculating the name over an over again, at the price of a really really tiny use case that probably isn't even noticed.